Sage/carbon
# feat(button, button-minor, button-toggle-group): add support for aria-labelledby and aria-describedby (FE-6364)

fixes #6567

### Proposed behaviour

- Add support for `aria-labelledby` and `aria-describedby` for `Button` and `ButtonMinor`
- Add support for `aria-describedby` for `ButtonToggleGroup` - accessible description is set on each toggle button inside the group ([as recommended by this source](https://devdoc.net/web/developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_group_role.html)).
- Simplify `Button` tests
- Update copyright dates

### Current behaviour

- Accessible description cannot be set on `Button` or `ButtonMinor`.[^1]
- Accessible description cannot be set on `ButtonToggleGroup`
- Copyright dates in README and licence are outdated

### Checklist




#### QA


### Testing instructions

Test with the following screen readers: **macOS VoiceOver (with Safari)** and **NVDA (with Chrome and Firefox)**.

- For `Button`:
  1. Open the test story "With External Labels".
  2. With a screen reader active, focus on the button.
  3. Verify if the accessible name "Add Product Addon" and description "You will not be charged until you checkout" are both read out.

- For `ButtonMinor`:
  1. Open the test story "With External Labels".
  2. With a screen reader active, focus on the button.
  3. Verify if the accessible name "Add Product Addon" and description "You will not be charged until you checkout" are both read out.
 
- For `ButtonToggle`:
  1. Open the story "with inputHint".
  2. With a screen reader active, focus on any of the three toggle buttons.
  3. Verify if the accessible name "inputHint example" and description "Hint text" are both read out.

[^1]: Possible to pass any HTML button attribute as a prop, but TypeScript throws an error as they're not officially part of either component's interfaces. https://github.com/Sage/carbon/blob/d0524b57701c790429dfa7e50361897e3841f552/src/components/button/button.component.tsx#L320
